src/controllers/project.js

```diff
@@ -46,6 +48,98 @@ const supportProject = async (req, res, next) => {
   return next();
 };
 
+const createReview = async (req, res, next) => {
+  const { id } = req.params;
+  const { userID, rating, content } = req.body;
+
+  try {
+    const project = await Project.findOne({ _id: id });
+
+    if (!project) return res.status(404).json({ message: 'Project not found' });
+
+    const review = {
+      user: userID,
+      rating,
+      content,
+    };
+
+    project.reviews.push(review);
+
+    await project.save();
+
+    res.json({
+      message: 'Review created successfully',
+      reviews: project.reviews,
+    });
+  } catch (error) {
+    res.status(422).json({ message: 'Unable to create review' });
+  }
+  return next();
+};
+
+const isEmpty = (field) => !!field;
```

Comment:
shouldn't it be a single negation?

Reply:
no this is not the usual NOT operator its the NOT NOT operator which converts the value to boolean readings more here => [https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_NOT#double_not_!!](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_NOT#double_not_!!)